chore: Update website in light of cloud (#2524)
electric-sql/electric: website/product/cloud.md

This allows you to scale out real-time data to [millions of concurrent users](/docs/reference/benchmarks#cloud) from a single commodity Postgres. With fast load times, low latency and consistent, low resource use.

## Usage

### Register your Database

1. Go to [Electric Cloud](https://dashboard.electric-sql.cloud) and log in.

2. Add a new database by clicking on [New Source](https://dashboard.electric-sql.cloud/sources/new).

3. Pick a region, team, and fill in your PostgreSQL connection string. Click the connect source button to connect your database to Electric Cloud.

Once connected you should see your source details akin to the screenshot below.

<img alt="Source details in cloud dashboard" src="/static/img/docs/cloud/source-details.png" />

It shouldn't take long before the source `state` becomes `active` and you're ready to make your first API request.

> Make sure to store the source ID and source secret credentials somewhere secure (like in your password manager) as you will need them to make API requests.

Reviewer: This isn't necessary given they can access the info whenever they want. This seems like a leftover from the old manual onboarding doc?

### Making API Requests

To request a shape you need to make an API request to `https://api.electric-sql.cloud/v1/shape`.
Don't forget to include the source credentials you obtained in the previous step.
If you don't recall them you can always find them in your user dashboard.

Here is an example request using `curl`:

```shell
export SOURCE_ID="8ea4e5fb-9217-4ca6-80b7-0a97581c4c10"

> export SOURCE_SECRET="<long secret value>"

Reviewer: Let's merge the PR making `secret` the default and update the docs here for the release

export SHAPE_DEFINITION="table=items&offset=-1"

curl -i "https://api.electric-sql.cloud/v1/shape?$SHAPE_DEFINITION\
    &source_id=$SOURCE_ID\
    &source_secret=$SOURCE_SECRET"
```

### Security Model

The source ID is a key that uniquely identifies your Postgres database.

The source secret is a token that grants access to it. You should treat the source secret as securely as you would with your database password.

> > [!Warning] Do not leak your source secret to the client

Reviewer: ```suggestion
> [!Warning] Do not use your source secret in the client!
```

> > If you use the source secret from an insecure client, then this exposes it to malicious users, who can then use it to connect to your cloud API.

Reviewer: ```suggestion
> If you use the source secret from a client, then this exposes it to malicious users, who can then use it to connect to your cloud API.
```

Let's just tell them to never use the secret in the client. There's no way of securing it really.
